source/core/constraint/unique.d

```diff
@@ -58,134 +54,111 @@ public class UniqueConstraint : Constraint
 	 * Ensures `cell` is connected to `cells` and that each Cell in `cells` is
 	 *     connected to `cell` \
 	 * A Cell cannot connect to itself \
-	 * Null values are not processed
+	 * Null values are not processed \
+	 * Duplicated objected are not processed
 	 *
 	 * Params:
 	 *     cell = cell to connect
 	 *     cells = cells to be connected
 	 *
-	 * SeeAlso: void createSingle(**cell1**,**cell2**)
+	 * SeeAlso: `@Cell` void addConstraint(**C : Constraint**)(**Cell[] cells...**)
 	 */
 	public static void createMultiSingle(Cell cell, Cell[] cells...)
 	{
 		if (cell is null)
 			return;
 
-		foreach (Cell c; cells)
-		{
-			if (c is null)
-				continue;
-
-			if (c != cell)
-				createSingle(cell, c);
-		}
+		cell.addConstraint!UniqueConstraint(cells);
 	}
 
 
 	/** Connects group of Cells to each other
 	 *
 	 * Ensures all Cells in 'cells' are connected to each other \
 	 * A Cell cannot connect to itself \
-	 * Null values are not processed
+	 * Null objects are not processed \
+	 * Duplicated objected are not processed
 	 *
 	 * Params:
 	 *     cells = cells to connect to each other
 	 *
-	 * SeeAlso: void createSingle(**cell1**,**cell2**)
+	 * SeeAlso: `@Cell` void addConstraint(**C : Constraint**)(**Cell[] cells...**)
 	 */
 	public static void createInterconnected(Cell[] cells...)
 	{
 		import std.array : array;
-		import std.algorithm : filter;
-		if (canFind(cells, null))
-		{
-			cells = cells.filter!(cell => cell !is null).array;
-		}
+		import std.algorithm : filter, uniq;
+		cells = uniq(cells.filter!(cell => cell !is null)).array;
 
 		foreach (Cell cell; cells)
 		{
-			createMultiSingle(cell, cells);
+			cell.addConstraint!UniqueConstraint(cells);
 		}
 	}
 
 
 	/** Removes a connection between two Cells
 	 *
 	 * A Cell cannot disconnect from itself \
-	 * Null values are not processed
+	 * Null values are not processed \
+	 * Duplicated objected are not processed
 	 *
 	 * Params:
 	 *     cell1 = cell to disconnect
 	 *     cell2 = cell to disconnect
+	 *
+	 * SeeAlso: `@Cell` void disconnect(**C : Constraint**)(**Cell[] cells...**)
 	 */
 	public static void removeSingle(Cell cell1, Cell cell2)
 	{
 		if (cell1 is null || cell2 is null || cell1 == cell2)
 			return;
 
 		cell1.disconnect!UniqueConstraint(cell2);
-		cell2.disconnect!UniqueConstraint(cell1);
 	}
 
 
 	/** Removes a connection between a Cell and a group of Cells
 	 *
 	 * A Cell cannot disconnect from itself \
-	 * Null values are not processed
+	 * Null values are not processed \
+	 * Duplicated objected are not processed
 	 *
 	 * Params:
 	 *     cell = cell to disconnect
 	 *     cells = cells to disconnect from
+	 *
+	 * SeeAlso: `@Cell` void disconnect(**C : Constraint**)(**Cell[] cells...**)
 	 */
 	public static void removeMultiSingle(Cell cell, Cell[] cells...)
 	{
-		foreach (c; cells)
-		{
-			if (c is null)
-				continue;
+		if (cell is null)
```

Comment:
Not sure if you're doing it in this project, but you could use contract programming here:

```d
public static void removeMultiSingle(Cell cell, Cell[] cells...)
in(cell != null)
{
    ...
}
```